projects/i18n/switch/load-lang.ts

```diff
@@ -0,0 +1,19 @@
+import {TuiLanguage, TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+import {from, Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+import {TuiLangChunkLoader} from './options';
+
+export function tuiLoadLanguage(
+    language: TuiLanguageName,
+    loader: Exclude<TuiLangChunkLoader, null | undefined>,
+): Observable<TuiLanguage> {
+    return from(normalizeCommonJSImport(loader(language))).pipe(
+        map((module: any) => module?.[`TUI_${language.toUpperCase()}_LANGUAGE`]),
+    ) as Observable<TuiLanguage>;
```

Comment:
You should be able to declare generic somewhere here without typecast.

Reply:
fixed